cm_sdk/models/__init__.py

```diff
@@ -0,0 +1,266 @@
+from enum import Enum
+import json
+from json import JSONEncoder
+import re
+import logging
+
+__author__ = 'e0d'
+
+
+class StringUtil(object):
+
+    class Case(Enum):
+        CAMEL = 1
+        SNAKE = 2
+
+    # credit
+    # http://stackoverflow.com/questions/19053707/convert-snake-case-snake-case-to-lower-camel-case-lowercamelcase-in-python
+    @classmethod
+    def to_camel_case(cls, snake_str):
+        components = snake_str.split('_')
+        return components[0].lower() + "".join(x.title() for x in components[1:])
+
+    @classmethod
+    def to_snake_case(cls, camel):
+        regex = re.compile(r'([A-Z])')
+        return regex.sub(r'_\1', camel).lower()
+
+    @classmethod
+    def convert_dict_keys(cls, dict_in, target_case=None):
```

Comment:
I'm uncomfortable with this approach because it makes the relationships between names in Python and names in JSON implicit. There's not anywhere you can go in the code where you can actually *see* how the names correspond to each other. I think it'd be better to have that mapping be explicit, just a list of tuples or something in each class. It's a little more verbose, but if you're generating it, I don't think it adds an unreasonable burden. Then, if the correspondence you want is camel->snake or anything else, it can be the responsibility of the generator.

Another advantage to this approach is that that mapping is a great place to stick metadata about more complex datatypes, e.g. datetimes so that they can be properly deserialized.